### PR TITLE
fix(eslint-plugin): [method-signature-style] don't auto-fix interfaces within namespaces

### LINKED ISSUE #2340: [method-signature-style] Using `property` breaks overloading for declared/global/merged types
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/method-signature-style": "error"
  }
}
```

I'm aware of this PR that fixed some overloading issues (https://github.com/typescript-eslint/typescript-eslint/pull/1966), but I went down another rabbit hole that took me a while to fix. When I upgraded eslint and enabled a bunch of rules, one of my projects type's completely broke.

I have this one implementation that declares a global to merge with some of Jest's types. I do this as I'm providing new Jest functionality.

```ts
declare global {
  namespace jest {
    interface Matchers<R, T> {
      toBeChecked(): R;
      toBeDisabled(): R;
      toBeElementType(type: ElementType): R;
      toBeNodeType(type: NodeType): R;
      toContainNode(node: NonNullable<React.ReactNode>): R;
      toHaveClassName(name: string): R;
      toHaveKey(value: string | number): R;
      toHaveProp<K extends keyof PropsOf<T>>(name: K, value?: PropsOf<T>[K]): R;
      toHaveProps(props: Partial<PropsOf<T>>): R;
      toHaveRendered(): R;
      toHaveValue(value: unknown): R;
    }
  }
}
```

I also provide more implementations in other packages, like so.

```ts
declare global {
  namespace jest {
    interface Matchers<R, T> {
      // Add overloads specific to the DOM
      toHaveProp<K extends keyof DomPropsOf<T>>(name: K, value?: DomPropsOf<T>[K]): R;
      toHaveProps(props: Partial<DomPropsOf<T>>): R;
    }
  }
}
```

This works, as TS is smart enough to inherit, merge, and provide overloads for all these methods (note that `toHaveProp` is declared in both examples). Once I enabled `method-signature-style`, I opted to use `property` since it's suggested when using strict. However, this style changed the above to:

```ts
declare global {
  namespace jest {
    interface Matchers<R, T> {
      // Add overloads specific to the DOM
      toHaveProp: <K extends keyof DomPropsOf<T>>(name: K, value?: DomPropsOf<T>[K]) => R;
      toHaveProps: (props: Partial<DomPropsOf<T>>) => R;
    }
  }
}
```

While this syntax isn't invalid on its own, it completely breaks the overloading within the `jest` namespace. I had to explicitly disable the `method-signature-style` rule for these instances.

**Expected Result**

Rule takes into account declars/globals/namespaces/merging, etc.

**Actual Result**

Overloads no longer existed, causing type errors across many call sites in the code base.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.3.0` |
| `@typescript-eslint/parser`        | `3.3.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `12.16` |
| `npm`                              | `6.14` |

### PR DESCRIPTION
**Pull request by aggmoulik**
Fixes #2340

Signed-off-by: Moulik Aggarwal <qwertymoulik@gmail.com>

### CODE REVIEW

**Comment by aggmoulik** [comment]
Fixes #2340

File: packages/eslint-plugin/src/rules/method-signature-style.ts [hunk 1]
```diff
@@ -97,6 +97,17 @@ export default util.createRule<Options, MessageIds>({
...
+    function isNodeParentModuleDeclaration(node: any): boolean {
+      if (node.parent?.type === AST_NODE_TYPES.TSModuleDeclaration) {
+        return true;
+      }
+
+      if (node.parent?.type === AST_NODE_TYPES.Program) {
+        return false;
+      }
+      return isNodeParentModuleDeclaration(node.parent);
+    }
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/method-signature-style.ts:109]: We can remove the `any` here with the catch-all `Node` type

```suggestion
    function isNodeParentModuleDeclaration(node: TSESTree.Node): boolean {
      if (!node.parent) {
        return false;
      }
    
      if (node.parent.type === AST_NODE_TYPES.TSModuleDeclaration) {
        return true;
      }

      if (node.parent.type === AST_NODE_TYPES.Program) {
        return false;
      }
      
      return isNodeParentModuleDeclaration(node.parent);
    }
```

File: packages/eslint-plugin/tests/rules/method-signature-style.test.ts [hunk 2]
```diff
@@ -348,5 +348,28 @@ interface Foo {
...
+      declare global {
+        namespace jest {
+          interface Matchers<R, T> {
+            // Add overloads specific to the DOM
+            toHaveProp<K extends keyof DomPropsOf<T>>(name: K, value?: DomPropsOf<T>[K]): R;
+            toHaveProps(props: Partial<DomPropsOf<T>>): R;
+          }
+        }
+      }
+      `,
```
Comment (bradzacher) [inline packages/eslint-plugin/tests/rules/method-signature-style.test.ts:362]: indent the example one line, and then you can use the fixer to auto-format the code block
```suggestion
      code: `
        declare global {
          namespace jest {
            interface Matchers<R, T> {
              // Add overloads specific to the DOM
              toHaveProp<K extends keyof DomPropsOf<T>>(name: K, value?: DomPropsOf<T>[K]): R;
              toHaveProps(props: Partial<DomPropsOf<T>>): R;
            }
          }
        }
      `,
```

**Review by bradzacher — CHANGES_REQUESTED** [review]
the logic itself LGTM - just two comments.
thanks for working on this!

**Comment by aggmoulik** [comment]
Hy @bradzacher , I have updated things you suggested.

**Review by bradzacher — APPROVED** [review]
LGTM - thanks for your contribution!

**Comment by aggmoulik** [comment]
Hy @bradzacher, Can you please add `hacktoberfest-accepted` label here.

**Comment by bradzacher** [comment]
I thought they only need a tag if they weren't going to be merged in October?

**Pull request merged**